From 07b3a944a52677c53df971ec5ef3419019b97f78 Mon Sep 17 00:00:00 2001
From: Erez Zadok <ezk@cs.sunysb.edu>
Date: Tue, 25 Dec 2007 16:55:17 -0500
Subject: [PATCH] Unionfs: set our superblock a/m/ctime granularity

Set it to 1 ns, because we could be stacked on top of file systems with such
granularity.

Signed-off-by: Erez Zadok <ezk@cs.sunysb.edu>
---
 fs/unionfs/main.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/fs/unionfs/main.c b/fs/unionfs/main.c
index 92f0e9d..23c18f7 100644
--- a/fs/unionfs/main.c
+++ b/fs/unionfs/main.c
@@ -642,6 +642,13 @@ static int unionfs_read_super(struct super_block *sb, void *raw_data,
 	/* max Bytes is the maximum bytes from highest priority branch */
 	sb->s_maxbytes = unionfs_lower_super_idx(sb, 0)->s_maxbytes;
 
+	/*
+	 * Our c/m/atime granularity is 1 ns because we may stack on file
+	 * systems whose granularity is as good.  This is important for our
+	 * time-based cache coherency.
+	 */
+	sb->s_time_gran = 1;
+
 	sb->s_op = &unionfs_sops;
 
 	/* See comment next to the definition of unionfs_d_alloc_root */
-- 
1.5.4.3

